llvm/test/CodeGen/RISCV/rvv/vwsll-vp.ll

```diff
@@ -238,12 +238,20 @@ define <vscale x 2 x i64> @vwsll_vx_i8_nxv2i64_zext(<vscale x 2 x i32> %a, i8 %b
 }
 
 define <vscale x 2 x i64> @vwsll_vi_nxv2i64(<vscale x 2 x i32> %a, <vscale x 2 x i1> %m, i32 zeroext %vl) {
-; CHECK-LABEL: vwsll_vi_nxv2i64:
-; CHECK:       # %bb.0:
-; CHECK-NEXT:    vsetvli zero, a0, e64, m2, ta, ma
-; CHECK-NEXT:    vzext.vf2 v10, v8
-; CHECK-NEXT:    vsll.vi v8, v10, 2, v0.t
-; CHECK-NEXT:    ret
+; CHECK-RV32-LABEL: vwsll_vi_nxv2i64:
+; CHECK-RV32:       # %bb.0:
+; CHECK-RV32-NEXT:    vsetvli zero, a0, e64, m2, ta, ma
+; CHECK-RV32-NEXT:    vzext.vf2 v10, v8
+; CHECK-RV32-NEXT:    vsll.vi v8, v10, 2, v0.t
```

Comment:
The transform did not apply on RV32 here:

    17373     if (!ISD::isConstantSplatVector(RHS.getNode(), ShAmt))
    (gdb)
    17374       return SDValue();
    (gdb) call RHS->dump()
    t20: nxv2i64 = splat_vector_parts Constant:i32<2>, Constant:i32<0>

I'm guessing it's an effect of type legalization. Please advise on how to fix.

Reply:
splat_vector_parts is used to splat an e64 element on rv32, i.e. when the EEW > XLEN. You could manually handle it in a follow up PR, I don't think it's critical for this PR

Reply:
Done. `combineOp_VLToVWOp_VL` handles `RISCVISD::VMV_V_X_VL` in `AfterLegalizeDAG` instead of `ISD::SPLAT_VECTOR_PARTS` in `AfterLegalizeTypes`. I do the same, so that `vwsll` has priority over `vwmulu`.